windows-agent/internal/proservices/proservices_test.go

```diff
@@ -64,6 +64,7 @@ func TestNew(t *testing.T) {
 			}
 
 			s, err := proservices.New(ctx, publicDir, privateDir, proservices.WithRegistry(reg))
+			defer s.Stop(ctx)
```

Comment:
You shouldn’t stop if the err is not nil (I didn’t recheck the code if it’s safe, but even if it is today, it might not be next time, so unfortunately, you need to check `if err == nil` which isn’t really idiomatic, I agree.